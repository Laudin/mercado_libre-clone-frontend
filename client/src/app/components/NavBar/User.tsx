import * as React from 'react'
import styled from 'styled-components/macro'

export function User() {
   return (
      <Wrapper>
         <Item><Link>Crear Cuenta</Link></Item>
         <Item><Link>Ingresar</Link></Item>
         <Item><Link>Mis Compras</Link></Item>
         <Item><Link>Carrito</Link></Item>
      </Wrapper>
   )
}

const Wrapper = styled.ul`
   display: flex;
   gap: 10px;
   list-style: none;
`
const Item = styled.li`
`
const Link = styled.a`
   color: black;
   text-decoration: none;
   cursor: pointer;
`
