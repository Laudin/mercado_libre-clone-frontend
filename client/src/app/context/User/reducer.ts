import { User } from '../../../types'

export const reducer = (state, action) => {
  switch (action.type) {
    case "set_user":
      localStorage.setItem('id', action.id)
      localStorage.setItem('name', action.name)
      localStorage.setItem('email', action.email)
      return {
        ...state,
        id: action.id,
        name: action.name,
        email: action.email
      }
    case "clear_user":
      localStorage.removeItem('id')
      localStorage.removeItem('name')
      localStorage.removeItem('email')
      return {
        ...state,
        id: '',
        name: '',
        email: ''
      }

    default:
      return state
  }
}

export const initialState: User = {
  id: localStorage.getItem('id') as string,
  name: localStorage.getItem('name') as string,
  email: localStorage.getItem('email') as string
}