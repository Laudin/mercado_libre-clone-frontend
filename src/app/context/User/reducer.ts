import { User } from '../../../types'

export const reducer = (state, action?) => {
  console.log(action)
  switch (action.type) {
    case "set_user":
      //localStorage.setItem('id', action.id)
      localStorage.setItem('name', action.name)
      localStorage.setItem('email', action.email)

      document.cookie = `id=${action.id}; max-age=${30 * 24 * 60 * 60}; Secure` //sets User Id 
      return {
        ...state,
        id: action.id,
        name: action.name,
        email: action.email
      }
    case "clear_user":
      //localStorage.removeItem('id')
      localStorage.removeItem('name')
      localStorage.removeItem('email')

      document.cookie = `id=${action.id}; expires=Thu, 01 Jan 1970 00:00:01 GMT"; max-age=${0}; Secure` // "kills" the cookie x_x
      return {
        ...state,
        id: '',
        name: '',
        email: ''
      }

    default:
      return state
  }
}

export const initialState: User = {
  // id: localStorage.getItem('id') as string,
  id: document.cookie.split(';').find((x: String) => x.match("id="))?.split("=")[1],
  name: localStorage.getItem('name') as string,
  email: localStorage.getItem('email') as string
}