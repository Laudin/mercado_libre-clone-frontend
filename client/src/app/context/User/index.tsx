import React from 'react';
import { reducer, initialState } from './reducer'

export const UserContext = React.createContext({
  currentUser: initialState,
  dispatch: reducer
})

export const UserProvider = ({ children }) => {
  
  const [currentUser, dispatch] = React.useReducer(reducer, initialState);
  
  return (
    <UserContext.Provider value={{currentUser, dispatch}}>
      {children}
    </UserContext.Provider>
  )
}