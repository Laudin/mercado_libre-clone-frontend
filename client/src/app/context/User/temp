export const reducer = (state, action: {type: string}) => {
  console.log(action)
  switch (action.type) {
    case "set_user":
      console.log('login')
      localStorage.setItem('name', state.name);
      localStorage.setItem('email', state.email);
      return {
        ...state,
        name: state.name,
        email: state.email
      }
    case "clear_user":
      console.log('logout')
      localStorage.removeItem('name');
      localStorage.removeItem('email');
      return {
        ...state,
        name: '',
        email: ''
      }
    default:
      return state
  }
}

export interface user {
  name: string,
  email: string
}

export const initialState: any = {
  name: '',
  email: ''
}

///////////////////////////////////

import React from 'react';
import { reducer, initialState, user } from './reducer'

export const UserContext = React.createContext({
  currentUser: initialState,
  dispatch: reducer
})

export const UserProvider = ({ children }) => {
  
  const [_, dispatch] = React.useReducer(reducer, initialState);

  const currentUser: user | any = () => {
    const name = localStorage.getItem('name') as string;
    const email = localStorage.getItem('email') as string;
    return {
      name: name,
      email: email
    };
  };
  /* console.log('currentUser: ')
  console.log(currentUser) */
  const setCurrentUser = () => {
    
  }
  
  return (
    <UserContext.Provider value={{currentUser, dispatch}}>
      {children}
    </UserContext.Provider>
  )
  /* const getUser = () => {
    const name = localStorage.getItem('name');
    const email = localStorage.getItem('email');
    //const userToken = JSON.stringify(tokenString as string); //pff alpedo
    //console.log(tokenString);
    return {
      name: name,
      email: email
    };
  };
   const saveUser = (name, email) => {
    localStorage.setItem('name', name);
    localStorage.setItem('email', email);
    setCurrentUser({
      name: name,
      email: email
    });
  }; */



  /* return {
    setCurrentUser: saveUser,
    currentUser,
  }; */
}