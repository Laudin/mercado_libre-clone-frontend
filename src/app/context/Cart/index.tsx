import React from 'react';
import { reducer, initialState } from './reducer'
import * as api from '../../api/usersApi'

export const CartContext = React.createContext({
   cart: initialState,
   dispatchCart: reducer
})

export const CartProvider = ({ children }) => {

   const [cart, dispatchCart] = React.useReducer(reducer, initialState);

   // infinite loop..
   // api.getCart().then(res => dispatchCart({ type: "initialize", product: '', initialValue: res }))

   return (
      <CartContext.Provider value={{ cart, dispatchCart }}>
         {children}
      </CartContext.Provider>
   )
}