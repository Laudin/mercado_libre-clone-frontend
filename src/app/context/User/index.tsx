import React from 'react';
import { reducer, initialState } from './reducer'

export const UserContext = React.createContext({
  currentUser: initialState,
  dispatchUser: reducer
})

export const UserProvider = ({ children }) => {

  const [currentUser, dispatchUser] = React.useReducer(reducer, initialState);

  return (
    <UserContext.Provider value={{ currentUser, dispatchUser }}>
      {children}
    </UserContext.Provider>
  )
}