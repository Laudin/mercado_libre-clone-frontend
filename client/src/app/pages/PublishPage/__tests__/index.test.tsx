import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import 'jest-styled-components'

import { PublishPage } from '../index';
import { BrowserRouter } from 'react-router-dom';
import { UserProvider } from 'app/context/User';

const shallowRenderer = createRenderer();

describe('<PublishPage />', () => {
   test('should render', () => {
      shallowRenderer.render(
         <BrowserRouter>
            <UserProvider>
               <PublishPage />
            </UserProvider>
         </BrowserRouter>
      );
      const rendererOutput = shallowRenderer.getRenderOutput();
      expect(rendererOutput).toMatchSnapshot()
   })
});