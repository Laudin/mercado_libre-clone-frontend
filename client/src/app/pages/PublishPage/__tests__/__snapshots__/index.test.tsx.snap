// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublishPage /> should match snapshot 1`] = `
<React.Fragment>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
    prioritizeSeoTags={false}
  >
    <title>
      Vender
    </title>
    <meta
      content="Publish new product to sell"
      name="publish"
    />
  </Helmet>
  <PublishPage__Wrapper>
    <h1>
      Completa el formulario del producto
    </h1>
    <PublishPage__Form
      id="Form"
      onSubmit={[Function]}
    >
      <PublishPage__Section1
        id="1"
      >
        <PublishPage__Label>
          Indicá tu producto, marca y modelo
        </PublishPage__Label>
        <p>
          Este será el título de tu producto. 
          <a
            href="#"
          >
            Cómo armar un buen título
          </a>
        </p>
        <PublishPage__Err />
        <PublishPage__Input
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Ej: Microondas Atma MD18"
        />
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section1>
      <PublishPage__Section
        id="2"
      >
        <PublishPage__Label>
          Seleccioná una categoría
        </PublishPage__Label>
        <p>
          Categoría sobre la que caerá tu producto. Las personas podrán encontrar tu producto bajo esta categoría.
          <a
            href="#"
          >
            ¿De qué categoría es mi producto?
          </a>
        </p>
        <PublishPage__Err />
        <PublishPage__Select
          name="category"
          onBlur={[Function]}
          onChange={[Function]}
        >
          <option
            value=""
          >
            --Seleccione una categoría--
          </option>
          <option
            value="Vehículos"
          >
            Vehículos
          </option>
          <option
            value="Inmuebles"
          >
            Inmuebles
          </option>
          <option
            value="Supermercado"
          >
            Supermercado
          </option>
          <option
            value="Tecnología"
          >
            Tecnología
          </option>
          <option
            value="Hogar y Muebles"
          >
            Hogar y Muebles
          </option>
          <option
            value="Electrodomésticos"
          >
            Electrodomésticos
          </option>
          <option
            value="Herramientas"
          >
            Herramientas
          </option>
          <option
            value="Construcción"
          >
            Construcción
          </option>
          <option
            value="Deportes y Fítness"
          >
            Deportes y Fítness
          </option>
          <option
            value="Moda"
          >
            Moda
          </option>
          <option
            value="Juegos y Juguetes"
          >
            Juegos y Juguetes
          </option>
          <option
            value="Bebés"
          >
            Bebés
          </option>
          <option
            value="Belleza y Cuidado Personal"
          >
            Belleza y Cuidado Personal
          </option>
          <option
            value="Salud y Equipamiento Médico"
          >
            Salud y Equipamiento Médico
          </option>
          <option
            value="Industrias y Oficinas"
          >
            Industrias y Oficinas
          </option>
          <option
            value="Agro"
          >
            Agro
          </option>
          <option
            value="Productos Sustentables"
          >
            Productos Sustentables
          </option>
          <option
            value="Servicios"
          >
            Servicios
          </option>
        </PublishPage__Select>
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section>
      <PublishPage__Section
        id="3"
      >
        <PublishPage__Label>
          Marca
        </PublishPage__Label>
        <PublishPage__Err />
        <PublishPage__Input
          name="brand"
          onBlur={[Function]}
          onChange={[Function]}
        />
        <PublishPage__Label>
          Modelo
        </PublishPage__Label>
        <p>
          Si no tienes el modelo de tu producto presiona N/A
        </p>
        <PublishPage__Err />
        <PublishPage__Container>
          <PublishPage__Input
            id="modelInput"
            name="model"
            onBlur={[Function]}
            onChange={[Function]}
          />
          <label>
            N/A
          </label>
          <PublishPage__Input
            id="checkbox"
            name="model"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            type="checkbox"
            value="N/A"
          />
        </PublishPage__Container>
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section>
      <PublishPage__Section
        id="4"
      >
        <PublishPage__Label>
          Es tu producto nuevo o usado?
        </PublishPage__Label>
        <PublishPage__Err />
        <PublishPage__Container>
          <label>
            Usado
          </label>
          <PublishPage__Input
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="Usado"
          />
        </PublishPage__Container>
        <PublishPage__Container>
          <label>
            Nuevo
          </label>
          <PublishPage__Input
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="Nuevo"
          />
        </PublishPage__Container>
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section>
      <PublishPage__Section
        id="5"
      >
        <PublishPage__Label>
          Sube 1 o más fotos
        </PublishPage__Label>
        <p>
          La primera foto será la portada. Es aconsejable que tenga un fondo blanco. 
          <a
            href="#"
          >
            ¿Cómo sacar una buena foto de tu producto?
          </a>
        </p>
        <PublishPage__Err />
        <PublishPage__Input
          name="photos"
          onBlur={[Function]}
          onChange={[Function]}
          type="file"
        />
        <div>
          <img
            src=""
          />
        </div>
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section>
      <PublishPage__Section
        id="6"
      >
        <PublishPage__Label>
          Cuánto costará tu producto?
        </PublishPage__Label>
        <PublishPage__Err />
        <PublishPage__Container>
          <label>
            $
          </label>
          <PublishPage__Input
            min={0}
            name="price"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="0.00"
            step={0.01}
            type="number"
          />
        </PublishPage__Container>
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section>
      <PublishPage__Section
        id="7"
      >
        <PublishPage__Label>
          Cuál es el stock de tu producto?
        </PublishPage__Label>
        <PublishPage__Err />
        <PublishPage__Input
          min={0}
          name="stock"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section>
      <PublishPage__Section
        id="8"
      >
        <PublishPage__Label>
          Una descripción de tu producto
          <p
            id="optional"
          >
            (opcional)
          </p>
        </PublishPage__Label>
        <textarea
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
        />
        <PublishPage__Next
          onClick={[Function]}
          type="button"
        >
          Siguiente
        </PublishPage__Next>
      </PublishPage__Section>
      <PublishPage__Section
        id="9"
      >
        <input
          type="submit"
        />
      </PublishPage__Section>
    </PublishPage__Form>
  </PublishPage__Wrapper>
</React.Fragment>
`;
